src/libstore/binary-cache-store.cc

```diff
@@ -515,8 +515,14 @@ void BinaryCacheStore::queryRealisationUncached(
             if (!data)
                 return (*callbackPtr)({});
 
-            auto realisation = Realisation::fromJSON(nlohmann::json::parse(*data), outputInfoFilePath);
-            return (*callbackPtr)(std::make_shared<const Realisation>(realisation));
+            std::shared_ptr<const Realisation> realisation;
+            try {
+                realisation = std::make_shared<const Realisation>(nlohmann::json::parse(*data));
+            } catch (Error & e) {
+                e.addTrace({}, "while parsing file '%s'", outputInfoFilePath);
```

Comment:
Does this include the expected format or type?
If not:

```suggestion
                e.addTrace({}, "while parsing file '%s' as a realisation", outputInfoFilePath);
```